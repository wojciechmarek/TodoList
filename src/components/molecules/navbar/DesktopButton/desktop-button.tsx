import { DesktopNavbarButtonProps } from './desktop-button.interface';
import {
  DesktopNavbarButtonContainer,
  DesktopNavbarButtonIcon,
  DesktopNavbarButtonText,
} from './desktop-button.styled';

export const DesktopNavbarButton = (props: DesktopNavbarButtonProps) => {
  const { icon, name, expanded, ...params } = props;
  return (
    <DesktopNavbarButtonContainer {...params}>
      <DesktopNavbarButtonIcon>{icon}</DesktopNavbarButtonIcon>
      <DesktopNavbarButtonText expanded={expanded}>
        {name}
      </DesktopNavbarButtonText>
    </DesktopNavbarButtonContainer>
  );
};
