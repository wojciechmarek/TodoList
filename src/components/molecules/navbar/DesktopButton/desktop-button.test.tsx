import { configure, render } from '@testing-library/react';
import { DesktopNavbarButtonProps } from './desktop-button.interface';
import { DesktopNavbarButton } from './desktop-button';
import '@testing-library/jest-dom';
import TestRenderer from 'react-test-renderer';

describe('DesktopNavbarButton', () => {
  let desktopNavbarButtonProps: DesktopNavbarButtonProps;

  beforeEach(() => {
    configure({ throwSuggestions: true });

    desktopNavbarButtonProps = {
      icon: <span>A</span>,
      expanded: false,
      name: 'Click me',
      onClick: jest.fn(),
    };
  });

  it('should render', () => {
    const { baseElement } = render(
      <DesktopNavbarButton {...desktopNavbarButtonProps} />
    );
    expect(baseElement).toBeTruthy();
  });

  it('should match the snapshot', () => {
    const tree = TestRenderer.create(
      <DesktopNavbarButton {...desktopNavbarButtonProps} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('when the desktop button is rendered', () => {
    it('should render the icon', () => {
      const { getByText } = render(
        <DesktopNavbarButton {...desktopNavbarButtonProps} />
      );
      const icon = getByText(/A/i);
      expect(icon).toBeTruthy();
    });

    it('should render the name', () => {
      const { getByText } = render(
        <DesktopNavbarButton {...desktopNavbarButtonProps} />
      );
      const name = getByText(/Click me/i);
      expect(name).toBeTruthy();
    });
  });

  describe('when the desktop button is clicked', () => {
    it('should call the onClick function', () => {
      const { getByText } = render(
        <DesktopNavbarButton {...desktopNavbarButtonProps} />
      );
      const button = getByText(/Click me/i);
      button.click();
      expect(desktopNavbarButtonProps.onClick).toHaveBeenCalled();
    });
  });
});
