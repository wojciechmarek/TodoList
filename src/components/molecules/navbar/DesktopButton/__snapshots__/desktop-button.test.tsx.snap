// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DesktopNavbarButton should match the snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  margin: 0.75rem;
  background-color: var(--navbar-settings-button-bg-color);
  width: calc(100% - 1.5rem);
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-transition: filter 0.3s ease-in-out;
  transition: filter 0.3s ease-in-out;
  border: var(--standard-border);
  border-radius: var(--standard-border-radius);
  box-shadow: var(--button-box-shadow);
  cursor: pointer;
}

.emotion-0:hover {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  color: var(--button-font-color);
  height: 2.5em;
  width: 2.5em;
}

.emotion-2 {
  height: calc(100% * 0.618);
  width: calc(100% * 0.618);
}

.emotion-3 {
  font-weight: 600;
  color: var(--button-font-color);
  display: none;
}

<div
  className="emotion-0"
  onClick={[MockFunction]}
>
  <div
    className="emotion-1"
  >
    <div
      className="emotion-2"
    >
      <span>
        A
      </span>
    </div>
  </div>
  <p
    className="emotion-3"
  >
    Click me
  </p>
</div>
`;
